### ISSUE TITLE
Implement `Input` for `Iterator<Item = Input>`

### ISSUE THREAD

**Issue by soenkehahn** [soenkehahn/cradle#101]
I recently wanted to pass `std::env::args().skip(1)` to a child process. So I did `std::env::args().skip(1).collect::<Vec<_>>()`. Maybe there should be an easier way to pass in iterators?

**Comment by casey** [comment]
I think we would need a wrapper type, since we can't due blanket impls due to the impl on `&I where I: Input`.

This works:

```rust
struct Inputs<I: IntoIterator<Item = T>, T: Input>(I);

impl<I: IntoIterator<Item = T>, T: Input> Input for Inputs<I, T> {
    fn configure(self, config: &mut Config) {
        for input in self.0 {
            input.configure(config);
        }
    }
}

#[test]
fn can_pass_multiple_inputs_with_inputs() {
    let crate::StdoutTrimmed(s) = crate::run_output!("echo", Inputs(vec!["a", "b", "c"]));
    assert_eq!(s, "a b c");
}
```

**Comment by soenkehahn** [comment]
Yeah, I was also thinking that a wrapper type might be necessary.

Out of curiosity, what exactly is the advantage of using `IntoIterator` over `Iterator` here?

**Comment by casey** [comment]
> Out of curiosity, what exactly is the advantage of using `IntoIterator` over `Iterator` here?

Usually, `Iterator` is implemented by a specialized iterator types, for example `std::vec::Iter` or `std::collections::hash_map::Keys`. `Vec` and `HashMap` themselves instead implement `IntoIterator` to return an iterator.
